modern-business/inc/fse-template-data.php

```diff
@@ -0,0 +1,324 @@
+<?php
+/**
+ * Template data inserter file.
+ *
+ * @package full-site-editing
+ */
+
+/**
+ * Class A8C_WP_Template_Data_Inserter
+ */
+class A8C_WP_Template_Data_Inserter {
```

Comment:
What if instead of including all of the logic in the theme, you instead keep the logic in the plugin and expose helper functions that the theme calls to register template parts, and provide the default block template content (or block template name to request from the API)?

Even though we are using one base theme that all child themes inherit from, that may not always be the case, and you're opening the possibility of future copy-pasta. Abstracting away the hard work, but providing helper functions matches the general philosophy of WordPress and theming.

Reply:
There's some discussion on this in the other PR, btw. :) 

https://github.com/Automattic/wp-calypso/pull/34640#issuecomment-511231921

Reply:
Ideally I'd like to do that and prevent the need for copy-pasta in the future. On the other hand I'm also trying to avoid posing the requirement that FSE plugin needs to be installed/active prior to theme activation - which might be problematic if the plugin code it relies on is missing.

Ideally I'd like FSE theme to be able to operate entirely independent of FSE plugin - there should be a fallback non-FSE header, and data population should work. That way, if the plugin is activated at a later time, FSE functionality will kick in immediately.